pkg/network/bridge/interfacecleaner.go

```diff
@@ -0,0 +1,66 @@
+package bridge
+
+import (
+	"sync"
+	"strconv"
+	
+	"github.com/fsnotify/fsnotify"
+	"github.com/vishvananda/netlink"
+)
+
+type interfaceCleaner struct {
+	hostInterfaceByPID map[int]string
+	mutex              *sync.Mutex
+}
+
+func newInterfaceCleaner() *interfaceCleaner {
+	return &interfaceCleaner{
+		hostInterfaceByPID: make(map[int]string),
+		mutex:              new(sync.Mutex),
+	}
+}
+
+func (ic *interfaceCleaner) Run() {
+	watcher, err := fsnotify.NewWatcher()
+	if err != nil {
+		panic(err)
+	}
+	defer watcher.Close()
+
+	err = watcher.Add("/proc")
+	if err != nil {
+		panic(err)
+	}
+
+	for {
```

Comment:
is there a graceful shutdown mechanism for infinite loop threads?

Reply:
We can pass a stop channel, but as for now, there is no one that would call it. We can add SIGNAL handling later.